Add release 1.20 releasenotes (#14101)
istio/istio.io: content/en/news/releases/1.20.x/announcing-1.20/_index.md

---
title: Announcing Istio 1.20.0
linktitle: 1.20.0
subtitle: Major Release
description: Istio 1.20 Release Announcement.
publishdate: 2023-11-14
release: 1.20.0
aliases:
- /news/announcing-1.20
- /news/announcing-1.20.0
---

> We are pleased to announce the release of Istio 1.20. This is the last Istio release of 2023. We would like to thank the

Reviewer: Is there a document with this file's content?  This is basically the marketing blog for the launch, and I'm happy to "punch these up" and make it more readable.

Author: Thanks! I'm going to create a one and paste the link here.

Author: @craigbox Here's the doc link: https://docs.google.com/document/d/1EO2cbtKYKH1eC4FS3o2l9pKANA308JdEbwbH6TTLblA/edit?usp=sharing

entire Istio community for helping get the 1.20.0 release published. We would like to thank the Release Managers for
this release, `Xiaopeng Han` from DaoCloud, `Aryan Gupta` from Google, and `Jianpeng He` from Tetrate. The release
managers would specially like to thank the Test & Release WG lead Eric Van Norman (IBM) for his help and guidance
throughout the release cycle. We would also like to thank the maintainers of the Istio work groups and the broader Istio
community for helping us throughout the release process with timely feedback, reviews, community testing and for all
your support to help ensure a timely release.

{{< relnote >}}

{{< tip >}}
Istio 1.20.0 is officially supported on Kubernetes versions `1.25` to `1.28`.
{{< /tip >}}

## What's new

### Gateway API

The Kubernetes [Gateway API](http://gateway-api.org/) is an initiative to bring a rich set of service networking APIs (
similar to those of Istio VirtualService and Gateway) to Kubernetes.

> Now Kubernetes Gateway API is GA, and Istio brings full support for it! This effort was a widespread community effort

Reviewer 2: ```suggestion
Now that the Kubernetes Gateway API is GA, and Istio brings full support for it! This effort was a widespread community effort
```

> The Kubernetes [Gateway API](http://gateway-api.org/) is an initiative to bring a rich set of service networking APIs (
> similar to those of Istio VirtualService and Gateway) to Kubernetes.
> 
> Now that the Kubernetes Gateway API is GA, and Istio brings full support for it! This effort was a widespread community effort
> across the broader Kubernetes ecosystem and has multiple conformant implementations (including Istio). This marks a
> significant milestone, as Istio users can now leverage the stable set of Gateway API
> features for enhanced traffic management and ingress control in production environments.
> 
> Check out the [mesh documentation](/docs/tasks/traffic-management/ingress/gateway-api/#mesh-traffic) to get started.

Reviewer 3: ```suggestion
The Kubernetes [Gateway API](http://gateway-api.org/) is an initiative to bring a rich set of service networking APIs
(similar to those of Istio VirtualService and Gateway) to Kubernetes.

Kubernetes [Gateway API is now GA](https://kubernetes.io/blog/2023/10/31/gateway-api-ga/)
and Istio provides [full support for it](https://gateway-api.sigs.k8s.io/implementations/#istio)!
This has been a widespread community effort
across the broader Kubernetes ecosystem that has produced multiple conformant implementations
(including [Istio's fully-conformant one](https://github.com/kubernetes-sigs/gateway-api/blob/main/conformance/reports/v1.0.0/istio-istio.yaml)).
This marks a significant milestone, as Istio users can now leverage the stable set of Gateway API
features for enhanced traffic management and ingress control in production environments.

Check out the [Gateway API task](/docs/tasks/traffic-management/ingress/gateway-api/) to get started.
```

In this release, we have also added support for configuring Istio
CRDs `AuthorizationPolicy`, `RequestAuthentication`, `Telemetry` and `WasmPlugin` for Kubernetes Gateway API via
the `targetRef` field.

> ### Ambient Mesh TODO(hanxiaop：ambient content and status)

Author: @howardjohn @linsun could you please check the `What's new` section and give me some suggestions on what needs to be added or removed? Also, I'm not sure about the ambient content. Thanks!

Author: @howardjohn @linsun Is there anything related to ambient that needs to be added in 1.20?

### Expanded Support for Network WasmPlugin

The extensibility of Istio is further broadened with support for network WasmPlugin with a new type `NETWORK`.

### TCP metadata exchange enhancements

Istio 1.20 brings two key updates to help control the TCP metadata exchange:

- **Fallback Metadata Discovery** Istio can now use a backup method to collect metadata. To use this, turn on
  the `PEER_METADATA_DISCOVERY` in the proxy and `PILOT_ENABLE_AMBIENT_CONTROLLERS` in the control plane.
- **ALPN Token Control**: There's a new setting called `PILOT_DISABLE_MX_ALPN` for the control plane. This lets you stop
  using a specific token `istio-peer-exchange` that's normally needed for services to talk to each other.

### Traffic Mirroring to Multiple Destinations

Traffic mirroring in Istio 1.20 now supports multiple destinations. This feature enables the mirroring of traffic to
various endpoints, allowing for simultaneous observation across different service versions or configurations.

### Plugged Root Cert Rotation

Security within Istio is improved through the added support for pluggable root certificate rotation.

> ### StartupProbe in Sidecar Containers

Reviewer 4: ```suggestion
### `StartupProbe` in Sidecar Containers
```

  using a specific token `istio-peer-exchange` that's normally needed for services to talk to each other.

### Traffic Mirroring to Multiple Destinations

Traffic mirroring in Istio 1.20 now supports multiple destinations. This feature enables the mirroring of traffic to
various endpoints, allowing for simultaneous observation across different service versions or configurations.

### Plugged Root Cert Rotation

Security within Istio is improved through the added support for pluggable root certificate rotation.

### StartupProbe in Sidecar Containers

To enhance pod startup times, Istio now includes a `startupProbe` in sidecar containers by default. This proactive
measure allows for aggressive polling during the initial phase without persisting throughout the pod's lifecycle,
potentially reducing startup times by an average of one second and improving overall resource efficiency.

### OpenShift Installation Enhancements

Istio's installation process on OpenShift clusters has been simplified, removing the need for granting the `anyuid`
SCC privilege to Istio and applications.

### Enhancements to the `istioctl` command

> Added a number of enhancements to the istioctl command including adding:

Reviewer 4: ```suggestion
Added a number of enhancements to the istioctl command including:
```
